saq/worker.py

```diff
@@ -102,6 +107,19 @@ def __init__(
         self.tasks: set[Task[t.Any]] = set()
         self.job_task_contexts: dict[Job, JobTaskContext] = {}
         self.dequeue_timeout = dequeue_timeout
+        self.burst = burst
+        self.max_burst_jobs = max_burst_jobs
+        self.burst_jobs_processed = 0
+        self.burst_jobs_processed_lock = threading.Lock()
```

Comment:
this needs to be an async lock, not a thread lock i think

Reply:
The `process` method is a sync one. Discussed in person